examples/softkiller/softkiller_plots.jl

```diff
@@ -0,0 +1,277 @@
+    event = select_ABS_RAP_max(event, rapmax)
+
+    distance = args[:distance]
+    algorithm = args[:algorithm]
+    p = args[:power]
+    strategy = args[:strategy]
+
+    filtered_event = PseudoJet[]
+    for (i, pseudo_jet) in enumerate(event)
+        # Reconstruct PseudoJet with cluster_hist_index for tracking
+        new_pseudo_jet = PseudoJet(JetReconstruction.px(pseudo_jet),
+                                   JetReconstruction.py(pseudo_jet),
+                                   JetReconstruction.pz(pseudo_jet),
+                                   JetReconstruction.energy(pseudo_jet);
+                                   cluster_hist_index = i)
+        push!(filtered_event, new_pseudo_jet)
+    end
+
+    # Clustering of the given vector of PseudoJets 
+    cluster_seq_pu = jet_reconstruct(filtered_event,
+                                     R = distance, p = p, algorithm = algorithm,
+                                     strategy = strategy)
+
+    finaljets_pu_lorv = inclusive_jets(cluster_seq_pu, ptmin = 25.0)
+    return finaljets_pu_lorv
+end
+
+# Helper to extract rapidity, phi, pt2, and color for plotting
+function push_data!(event::AbstractVector, y::Vector{Float64}, phi::Vector{Float64},
+                    pt::Vector{Float64}, colors::Vector{String},
+                    color::String, origin::Dict{PseudoJet, String})
+    for jet in event
+        pj = isa(jet, PseudoJet) ? jet :
+             PseudoJet(JetReconstruction.px(jet), JetReconstruction.py(jet),
+                       JetReconstruction.pz(jet), JetReconstruction.energy(jet))
+        push!(y, JetReconstruction.rapidity(pj))
+        push!(phi, JetReconstruction.phi(pj))
+        push!(pt, JetReconstruction.pt2(pj))
+        # Color code by origin (hard/pileup) if available
+        if haskey(origin, jet)
+            push!(colors, origin[jet] == "hard" ? "purple" : "white")
+        else
+            push!(colors, color)
+        end
+    end
+end
+
+# Plot a scatter of jets or particles in (y, phi) with pt-dependent marker size
+function plot_set_up(y::Vector{Float64}, phi::Vector{Float64}, pt::Vector{Float64},
+                     color::Vector{String}, plot_title::String)
+    fig = Figure()
+    ax = Axis(fig[1, 1],
+              xlabel = "Rapidity (y)", ylabel = "Azimuthal Angle (φ)",
+              title = plot_title,
+              limits = ((-5.0, 5.0), (0.0, 6.5)))
+
+    marker_sizes = 8 .+ pt ./ 50
+    color_syms = Symbol.(color)
+
+    scatter!(ax, y, phi;
+             markersize = marker_sizes,
+             strokecolor = :black,
+             strokewidth = 1.0,
+             color = color_syms,
+             transparency = true)
+
+    Legend(fig[1, 2],
+           [
+               PolyElement(color = :white),
+               PolyElement(color = :purple),
+               PolyElement(color = :royalblue3)
+           ],
+           ["Pileup", "Hard Event", "Jet"], "Legend")
+
+    save_dir = "examples/softkiller"
```

Comment:
Assume that we're running in `examples/softkiller`, so you don't need this path.

Also, make sure there's a `.gitignore` to not accidentally commit the output PNG files.